lib/public/views/Runs/Runs.js

```diff
@@ -196,12 +196,37 @@ export default class Overview extends Observable {
 
     /**
      * Filter runs based on the selected run numbers and fields to prepare it for export
-     * @param {Object} runs The runs data
      * @returns {Object} The Filtered runs data
      */
-    async getFilteredRuns(runs) {
-        const selectedRuns = runs.filter((run) => this.getSelectedRunNumbers().includes(run.runNumber));
-        const filteredRuns = selectedRuns.map((selectedRun) => pick(selectedRun, this.getSelectedRunsFields()));
+    async getFilteredRuns() {
+        this.runs = RemoteData.loading();
+        this.notify();
+
+        const runIds = this.getSelectedRunNumbers().split(',');
+        const foundRuns = [];
+        let response;
+        let result;
+
+        // Loops through every run id that has been filled in and retrieves them from the database
+        for (let i = 0; i < runIds.length; i++) {
```

Comment:
Usage of `1 letter` variables is discouraged.
I would recommend you adapt the `for loop` to make use of either:
* [Array.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) which will retrieve you an array already, perfect for your use case of `foundRuns` and you can also chain it with `filteredRuns` from the end of the method
* [Array.forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)

Reply:
Array.map and Array.forEach are both synchronous functions, I tried running it like this and a couple other similar variations:
- await runIds.map(async runID => { ... })

But they all returned lists of promises.

So I'll change the for loop into a for-of loop which can be used async and give the variable a proper name.

Reply:
Indeed, it will return a list of promises. 

Just as a learning exercise you could probably make use of `Promise.all()` in combination with the map. There is an nice example here: https://advancedweb.hu/how-to-use-async-functions-with-array-map-in-javascript/

No need to update the code thought, the `for of` is good enough.